### PR TITLE
Initial repository setup

### PR DESCRIPTION
**Pull request by cameronwaterman**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/systemlink-operations-handbook/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- First pass at README files
- Basic application structure
- Skeleton for usage documentation
- .tox file for automation
- Test infrastructure

### Why should this Pull Request be merged?

Initial repository setup requirements.

### What testing has been done?

- Ran tox - outcome as expected
- Links in .md files work

### CODE REVIEW

File: Makefile
Comment (spanglerco) [inline Makefile]: Any particular reason to use make instead of something like tox?

File: docs/conf.py
Comment (spanglerco) [inline docs/conf.py]: I believe NI owns the copyright. We use this setting in the Python client repo:

```suggestion
copyright = f'2020-{datetime.date.today().year}, National Instruments'
author = 'National Instruments'
```

File: setup.py
Comment (spanglerco) [inline setup.py]: Based on the Python clients:

```suggestion
    author="National Instruments",
    maintainer="Cameron Waterman",
    maintainer_email="cameron.waterman@ni.com",
```

File: setup.py
Comment (spanglerco) [inline setup.py]: Duplicate

File: .gitignore
Comment (spanglerco) [inline .gitignore]: General comment, have we considered using https://github.com/ni/nisystemlink-clients-python as a starting point for all of this?

**Review by spanglerco — CHANGES_REQUESTED** [review]

**Comment by spanglerco** [comment]
Looks like the repo isn't configured to require PR approvals before merging yet :)

**Comment by cameronwaterman** [comment]
> Looks like the repo isn't configured to require PR approvals before merging yet :)

Yes, I'll configure that. Related, I accidentally pushed the initial README.md and CONTRIBUTING.md directly to master. So please take a look at the existing content of those files as well.

File: Makefile
Comment (cameronwaterman) [reply Makefile]: Most of the Python project structure guidelines and example projects I found were using make (Including Mark's, which I pulled this Makefile from). For what it's worth, I _really_ would like to avoid make. I'll take a look into tox, or some other alternative and respond back to this conversation afterwards,

File: docs/conf.py
Comment (cameronwaterman) [reply docs/conf.py]: Well that makes sense. Is it still National Instruments, or just NI now? Not sure since this is a legal-related.

File: .flake8
Comment (spanglerco) [inline .flake8]: We can probably drop this unless we find a need for it.
```suggestion
```

File: setup.py
Comment (spanglerco) [inline setup.py]:
```suggestion
        version = "0.1.0"
```

File: setup.py
Comment (spanglerco) [inline setup.py]: I believe the client repo only needed mypy as a tests require because it would invoke it for some of the tests.

```suggestion
    tests_require=["pytest", "pytest-asyncio"],
```

**Review by spanglerco — CHANGES_REQUESTED** [review]
> What testing has been done?

Can you update your PR description, hopefully to say that you ran `tox` :)

File: tox.ini
Comment (spanglerco) [inline tox.ini]: Sorry for the misunderstanding, but we do want to run mypy from tox. My comment was that in the python client repo, there were pytests that also ran mypy during the tests, which I thought is why it shows up in tests_require given that tox has its own deps list.

Comment (cameronwaterman) [reply tox.ini]: Ah I see, my bad. I thought we might want it but figured you had a reason for removing it. That makes much more sense. I'll update 😄

**Review by spanglerco — APPROVED** [review]

**Pull request merged**
